src/main/java/com/googleinterns/zoomtube/servlets/DiscussionServlet.java

```diff
@@ -64,34 +65,27 @@ public void init() throws ServletException {
   public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
     long lectureId = Long.parseLong(request.getParameter(PARAM_LECTURE));
     Key lecture = KeyFactory.createKey(LectureUtil.KIND, lectureId);
-    String parentIdString = request.getParameter(PARAM_PARENT);
-    Key parent = null;
-    if (parentIdString != null) {
-      long parentId = Long.parseLong(parentIdString);
-      parent = KeyFactory.createKey(CommentUtil.KIND, parentId);
-    }
     User author = userService.getCurrentUser();
     if (author == null) {
       response.sendError(HttpServletResponse.SC_FORBIDDEN, "You are not logged in.");
       return;
     }
     String content = CharStreams.toString(request.getReader());
-
-    // Default timestamp is the start of the video.
-    long timestampMs = 0;
-    if (request.getParameter(PARAM_TIMESTAMP) != null) {
-      timestampMs = Long.parseLong(request.getParameter(PARAM_TIMESTAMP));
-    }
-
     Date dateNow = new Date(Clock.systemUTC().millis());
 
-    if (parent == null) {
-      datastore.put(CommentUtil.createEntity(lecture, timestampMs, author, content, dateNow));
+    final Entity commentEntity;
+    Comment.Type type = Comment.Type.valueOf(request.getParameter(PARAM_TYPE));
+    if (type == Comment.Type.REPLY) {
+      long parentId = Long.parseLong(request.getParameter(PARAM_PARENT));
```

Comment:
Same here

Reply:
I don't think that the parent would be null if the comment is a reply. I think the if statement accounts for that.

Reply:
Added TODO for separate PR, see #178